### DISCUSSION TITLE
when consumer is failing to connect the queue add interval for reconnection attempts

### DISCUSSION THREAD

**Discussion by tzilaswimmer** [bbc/sqs-consumer#482] [Feature Request]
### Background

I see that using the listener of on error make the consumer to try and reconnect every 10 seconds. I would prefer to configure this interval. Like `heartbeatInterval`, `pollingWaitTimeMs` etc. Maybe it is a good thing to have max attempts, but the interval is more important at the moment in my opinion

### Objectives

1. make reconnects more configurable
2.
3.

**Comment by tzilaswimmer** [comment]
Hi, 
What do you think?

**Comment by nicholasgriffintn** [comment]
Hey, sorry I missed this, we have an authentication timeout that's configurable for connection errors.

We're not really planning to do this for any errors.

**Reply by tzilaswimmer** [reply]
Hi @nicholasgriffintn thanks for the response.
We are experiencing many connection errors because the system attempts to reconnect every 10 seconds. This means it continually tries to connect, establishes a connection, encounters a network error, and repeats the process, resulting in an exception every 10 seconds. We aim to reduce the interval between connection attempts. Since I am not familiar with the architecture, would it be problematic to add intervals for other connection errors as well?
